openghg/retrieve/_access.py

```diff
@@ -410,24 +415,25 @@ def get_obs_surface_local(
         data = ds_resampled
 
     # Rename variables
-    rename: dict[str, str] = {}
-
-    data_variables = [str(v) for v in data.variables]
-    for var in data_variables:
-        if var.lower() == species.lower():
-            rename[var] = "mf"
-        if "repeatability" in var:
-            rename[var] = "mf_repeatability"
-        if "variability" in var:
-            rename[var] = "mf_variability"
-        if "number_of_observations" in var:
-            rename[var] = "mf_number_of_observations"
-        if "status_flag" in var:
-            rename[var] = "status_flag"
-        if "integration_flag" in var:
-            rename[var] = "integration_flag"
-
-    data = data.rename_vars(rename)  # type: ignore
+    if rename_vars:
+        rename: dict[str, str] = {}
+
+        data_variables = [str(v) for v in data.variables]
```

Comment:
We should employ pattern matching here, instead of conditional statements. 
for e.g variable_patterns = {"repeatability":"mf_repeatability", "variability":"mf_variability", ...} and loop and break for match. This will ensure we are increasing readability and scalability in future if required, whilst not repeating.

Reply:
I wouldn't want to make that change on this PR as this is just to allow these steps to be skipped but makes no other changes. Updating how the data variables names are updated should really be a separate issue.

Reply:
I understand we add a check and ensure the steps are skipped, However, losing track of such code-refining changes is easier. Thus if changes like these are small and don't impact the functionality hugely, it is worth making the change.

Reply:
If you wanted to put in a separate PR for this you'd be welcome to but I think we'd be best to keep this PR simple and within the scope of the original idea.

Reply:
Resolving and referenced in https://github.com/openghg/openghg/issues/1141